indexreport.go

```diff
@@ -35,15 +36,30 @@ type IndexReport struct {
 	Err string `json:"err"`
 }
 
+var (
+	zeroDist Distribution
+	zeroRepo Repository
+)
+
 // IndexRecords returns a list of IndexRecords derived from the IndexReport
 func (report *IndexReport) IndexRecords() []*IndexRecord {
 	out := []*IndexRecord{}
-	for _, pkg := range report.Packages {
-		for _, env := range report.Environments[pkg.ID] {
-			record := &IndexRecord{}
-			record.Package = pkg
-			record.Distribution = report.Distributions[env.DistributionID]
-			record.Repository = report.Repositories[env.RepositoryID]
+	for pkgID, pkg := range report.Packages {
+		for _, env := range report.Environments[pkgID] {
+			record := &IndexRecord{
+				ID:      pkgID,
```

Comment:
This seems odd... The IndexRecord is just a container, why would you float the pkgID to a root field ?